debezium-connector-mysql/src/main/java/io/debezium/connector/mysql/SnapshotReader.java

```diff
@@ -310,7 +310,12 @@ protected void execute() {
                 if (!snapshotLockingMode.equals(MySqlConnectorConfig.SnapshotLockingMode.NONE) && useGlobalLock) {
                     try {
                         logger.info("Step 1: flush and obtain global read lock to prevent writes to database");
-                        sql.set("FLUSH TABLES WITH READ LOCK");
+                        if (snapshotLockingMode.equals(MySqlConnectorConfig.SnapshotLockingMode.MINIMAL_PERCONA)) {
```

Comment:
How about providing the statement from a method on the `SnapshotLockingMode` enum?